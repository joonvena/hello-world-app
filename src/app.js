import express from 'express';
import { resolve } from 'path';
import bodyParser from 'body-parser';
import views from './views';
import api from './api';
import storage from './storage';

const app = express();
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));
app.use((req, res, next) => {
  console.log('Time:', new Date().toISOString(), ', Url: ', req.url);
  next();
});

app.use('/api', api(storage, GIT_COMMIT, GIT_TAG));
app.use('/', views(storage));

app.use('/assets', express.static(resolve(__dirname, 'assets')));

app.close = () => { };

export default app;
