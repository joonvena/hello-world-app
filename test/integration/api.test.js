
import supertest from 'supertest';
import app from '../../src/app';

const request = supertest(app);

afterAll(() => app.close());

describe('/api/version', () => {
  describe('GET', () => {
    it('should return version', async () => {
      const { body } = await request.get('/api/version').expect(200);

      expect(body).toMatchSnapshot({
        commit: expect.any(String),
        tag: expect.any(String),
        hostIp: expect.any(String),
        namespace: expect.any(String),
        podName: expect.any(String),
        podUid: expect.any(String),
        startedAt: expect.any(String),
      });
    });
  });
});

describe('/api/healthy', () => {
  describe('GET', () => {
    it('should return ok', async () => {
      const { body } = await request.get('/api/healthy').expect(200);

      expect(body).toMatchSnapshot();
    });
  });
});
