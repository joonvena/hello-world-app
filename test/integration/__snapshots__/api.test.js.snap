// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/healthy GET should return ok 1`] = `
Object {
  "status": "ok",
}
`;

exports[`/api/version GET should return version 1`] = `
Object {
  "commit": Any<String>,
  "hostIp": Any<String>,
  "namespace": Any<String>,
  "podIp": "unknown",
  "podName": Any<String>,
  "podUid": Any<String>,
  "startedAt": Any<String>,
  "tag": Any<String>,
}
`;
